### ISSUE TITLE
Flaky tests: org.eclipse.jgit.errors.MissingObjectException

### ISSUE THREAD

**Issue by bzz** [src-d/gemini#20]
On local `./sbt run` as well as on CI, sometimes test become flaky with `org.eclipse.jgit.errors.MissingObjectException`.

Example:

```
ERROR 15:13:13,185 org.apache.spark.internal.Logging$class (Logging.scala:91) - Exception in task 0.0 in stage 1.0 (TID 2)
org.eclipse.jgit.errors.MissingObjectException: Missing commit 4aa29ac236c55ebbfbef149fef7054d25832717f
	at org.eclipse.jgit.internal.storage.file.WindowCursor.open(WindowCursor.java:164)
	at org.eclipse.jgit.revwalk.RevWalk.getCachedBytes(RevWalk.java:903)
```

CI
 - https://api.travis-ci.org/v3/job/319232107/log.txt from [this run](https://travis-ci.org/src-d/gemini/builds/319232106), but same commit [passes here](https://travis-ci.org/src-d/gemini/jobs/319209976)
 - TODO add more
 - 

This might, or might not have to do something either with Engine behaviour, or with test fixtures that we have.

This issue is about investigating the reason and if needed, filing appropriate issues elsewhere to fix it.

**Comment by smacker** [comment]
Currently, we have 4 fixtures as siva files.
I can reproduce the bug without running tests but just hash command.

First I tried to remove all fixtures and recover only one. Then run `hash` command multiple times.
To distinguish bugs of engine & fixtures. If I can reproduce for every fixture - it means the problem is in the engine and doesn't depend on fixtures. If not, the problem is engine + particular fixture.

```sh
#!/bin/bash

set -e

i="0"
while [ $i -lt 1000 ]
do
    ./hash $1
    i=$[$i+1]
done
```

Can reproduce with:
2636f3c62f1a407b2996da6e3fe6fdc5d1ccd764.siva
5fb38a5744b2496ff8484e57b46a754433ede457.siva
9279be3cf07fb3cca4fc964b27acea57e0af461b.siva

Error: `org.eclipse.jgit.errors.MissingObjectException: Missing commit <sha1>`

Can **not** reproduce with:
f281ab6f2e0e38dcc3af05360667d8f530c00103.siva

**Comment by smacker** [comment]
Next, I tried to update engine to v0.2 and see if I still can reproduce the bug.
I chose to use v0.2 instead of v0.3.1 because v0.3 requires repository format and it will cause larger changes in gemini code.

With 0.2 we actually don't need to join anymore, but I kept it to for testing purposes.
With `2636f3c62f1a407b2996da6e3fe6fdc5d1ccd764.siva` I still can get `Missing commit` error.
Then I removed join (all necessary staff is already in getBlobs column), without join it works as expected.
